tests/compile-fail/interned_not_update.rs

```diff
@@ -0,0 +1,17 @@
+use salsa::Database as Db;
+
+#[salsa::input]
+struct MyInput {}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct NotUpdate<'db>(&'db ());
+
+#[salsa::tracked]
+fn tracked_fn<'db>(_db: &'db dyn Db, _: (), _: NotUpdate<'db>) {}
```

Comment:
Nit, I would probalby just use a `&str` here, it should be sufficient for demonstration purposes

```suggestion
fn tracked_fn<'db>(_db: &'db dyn Db, _: (), _not_update: &'db str) {}
```